Adding missing release notes (#2789)
kiegroup/kie-docs: doc-content/jbpm-docs/src/main/asciidoc/ReleaseNotes/Release.7.42.0.Final/immutable-springboot.adoc

[id='immutable-springboot-742']

= Immutable SpringBoot Deployments

> When building a SpringBoot application which embeds the jBPM process engine, the necessary processes (and rules, forms, etc.), bundled in a kjar, can be either embedded as part of the application or loaded on the fly.  In cloud environments, immutable applications are often wanted, and we have therefore improved our kie-maven-plugin to more easily include all necessary kjars (and dependencies) as part of the application so it is self-contained, by adding a new package-dependencies-kjar goal.

Reviewer: ```suggestion
When building a SpringBoot application that embeds the jBPM process engine, the necessary processes (rules, forms, and so in), bundled in a kjar, can be either embedded as part of the application or loaded on the fly. In cloud environments, immutable applications are often wanted, and we have therefore improved our kie-maven-plugin that includes all necessary kjars (and dependencies) as part of the application so it is self-contained, by adding a new package-dependencies-kjar goal.
```
